torchx/schedulers/kueue_job_scheduler.py

```diff
@@ -0,0 +1,809 @@
+
+        try:
+            api_instance = self._batchv1_api()
+            job = api_instance.read_namespaced_job_status(name, namespace)
+        except client.ApiException as e:
+            return f"Exception: {e}"
+        
+
+        try: 
+            status = job.status
+        except:
+            status = None
+        
+        if status:
+            condition = status.conditions[0]
+            if condition.type == "Suspended" and condition.reason == "JobSuspended":
+                app_state = JOB_STATE["Suspended"]
+            elif condition.reason == "JobResumed":
+                if status.active is not None:
+                    app_state = JOB_STATE["Running"]
+                elif status.succeeded is not None:
+                    app_state = JOB_STATE["Completed"]
+                elif status.failed is not None:
+                    app_state = JOB_STATE["Failed"]
+                else:
+                    app_state = JOB_STATE["Pending"]
+            else:
+                app_state = JOB_STATE["Pending"]
+
+            for condition in status.conditions or []:
+                role, _, idx = condition.type.rpartition("-")
+                state_str = condition.reason
+                if state_str == "JobResumed":
+                    state = KUEUE_STATE[state_str]
+                elif condition.type == "Complete":
+                    state = KUEUE_STATE["Complete"]
+                else:
+                    state = KUEUE_STATE["Unknown"]
+
+                if role not in roles:
+                    roles[role] = Role(name=role, num_replicas=0, image="")
+                    roles_statuses[role] = RoleStatus(role, [])
+
+                roles[role].num_replicas += 1
+                roles_statuses[role].replicas.append(
+                    ReplicaStatus(id=0, role=role, state=state, hostname="")
+                )
+        else:
+            app_state = AppState.UNKNOWN
+
+        return DescribeAppResponse(
+            app_id=app_id,
+            roles=list(roles.values()),
+            roles_statuses=list(roles_statuses.values()),
+            state=app_state,
+        )
+
+    def log_iter(
+        self,
+        app_id: str,
+        role_name: str,
+        k: int = 0,
+        regex: Optional[str] = None,
+        since: Optional[datetime] = None,
+        until: Optional[datetime] = None,
+        should_tail: bool = False,
+        streams: Optional[Stream] = None,
+    ) -> Iterable[str]:
+        assert until is None, "kubernetes API doesn't support until"
+
+        if streams not in (None, Stream.COMBINED):
+            raise ValueError(
+                "KueueJobScheduler only supports COMBINED log stream"
+            )
+
+        from kubernetes import client, watch
+
+        namespace, name = app_id.split(":")
+
+        pod_name = get_pod_name_from_job(self, job_name=name, namespace=namespace)
+        if pod_name is None:
+            return "Pods not found. Is the Job Suspended?"
```

Comment:
I don't think returning a string is appropriate here. I would either raise an error or log an exception and return an empty list.